### ISSUE TITLE
How to build and design blocks for consistent dev and user experiences

### ISSUE THREAD

**Issue by bph** [WordPress/developer-blog-content#350]
### Discussed in https://github.com/WordPress/developer-blog-content/discussions/346

<div type='discussions-op-text'>

<sup>Originally posted by **justintadlock** November 21, 2024</sup>
We should have a good guide for building blocks so that they:

- Provide a nice user experience that is consistent with Core design principles
- Make it easy for theme authors to integrate with them
- Favors smaller pieces that can be integrated into patterns
- Integrate with Core design tools
- Use Core components as much as possible
- When a binding, variation, or style variation would be better

Basically, this should be a sort of "dos" and "don'ts" of block building with perhaps some tips/tricks thrown in. Mostly, I'm just throwing out some ideas at the moment. Please contribute to the discussion.

This idea came out of a discussion around WooCommerce and creating a consistent user experience. CC: @ellenbauer</div>

**Comment by bph** [comment]
@justintadlock Would you be available again to continue working on this article?

**Comment by justintadlock** [comment]
I'll leave this open for other writers at the moment.

**Comment by AleksandrSamokhin** [comment]
I want to write this article. Should I jump in?

**Comment by bph** [comment]
Yes certainly @AleksandrSamokhin What do you need to get started?
